### ISSUE TITLE
Make "Precursor warning" more case-specific?

### ISSUE THREAD

**Issue by schymane** [MassBank/RMassBank#119]
It seems we are now getting the warning that was added in response to #102 in cases where the "fillPrecursorScan=FALSE" is not the correct cause. For example, I received the following when I processed a lot of files when the precursor was found in most cases, but not all (because we were fiddling around - but there are of course cases where the precursor is not detected at all).
Warning messages:
1: In findMsMsHR.mass(f, row[[1]], 0.5, ppm(row[[1]], 10, p = TRUE),  :
  No precursor was detected. It is recommended to try to use the setting fillPrecursorScan: TRUE in the ini-file
2: In findMsMsHR.mass(f, row[[1]], 0.5, ppm(row[[1]], 10, p = TRUE),  :
  No precursor was detected. It is recommended to try to use the setting fillPrecursorScan: TRUE in the ini-file
3: In findMsMsHR.mass(f, row[[1]], 0.5, ppm(row[[1]], 10, p = TRUE),  :
  No precursor was detected. It is recommended to try to use the setting fillPrecursorScan: TRUE in the ini-file

Solutions?
a) change the warning so that it e.g. reads:
No precursor was detected. Please check the mass and retention time window. If this happens for all cases, try using the setting fillPrecursorScan: TRUE in the ini-file
b) do some sort of check of the cause before printing the warning?
Any other ideas?

**Comment by ermueller** [comment]
Well, we could check the cause, but I guess the cause in most cases would mostly be "because of the data", in one way or another, right? And that would be outside of anything we can do.

I'd rather use the better error message.

**Comment by meowcat** [comment]
Hm the thing is, is this really an error? Or rather, what was the exact reason why the warning was implemented? It will warn you on every compound where no spectrum is found, right? Do we want this?

**Comment by schymane** [comment]
It isn’t necessarily an error, you’re right.
Erik added this to help point people towards the case that we needed the fillPrecursorScan=TRUE. This would mean we miss EVERY substance. The downside is that we now get this warning for everything we miss. For the case I noticed, we only missed a few, which we’d expect in a normal run especially if we e.g. don’t know all substances. It’s not a bad thing to have the warning there, I’d agree with Erik we should just improve the message.
This was another thing we added to our script after, to check if we actually found anything :-)

I’d vote that we actually print out the compound ID and maybe even the mass with the warning, this would help users know if they just messed up the SMILES, or they have a substance in positive that they’d only really expect in negative, or they got the wrong adduct (quaternary ammoniums?) or if it’s really every single substance. If one then gets this error for all substances then maybe the settings need to change …

**Comment by ermueller** [comment]
> I’d vote that we actually print out the compound ID and maybe even the mass with the warning, this would help users know if they just messed up the SMILES, or they have a substance in positive that they’d only really expect in negative, or they got the wrong adduct (quaternary ammoniums?) or if it’s really every single substance. If one then gets this error for all substances then maybe the settings need to change …

Yeah, that sounds good, printing the ID and mass. Should I do it this way?

**Comment by schymane** [comment]
Yes from my side!

**Comment by schymane** [comment]
according to Erik fixed but not yet merged

**Comment by meowcat** [comment]
Merged in the refactoring branch. Note though that I made the cpdID parameter in findMsMsHR.mass optional because we need the function a lot for cases where we actually go by mass and have no compound ID. Also reflected this in the doc. Will close once refactoring is merged into master.
